cmd/tink-worker/action.go

```diff
@@ -43,79 +57,81 @@ func executeAction(ctx context.Context, action *pb.WorkflowAction, wfID string)
 		timeCtx, cancel = context.WithTimeout(context.Background(), 1*time.Hour)
 	}
 	defer cancel()
-	//run container with timeout context
-	//startedAt := time.Now()
-	err = runContainer(timeCtx, id)
+	err = runContainer(timeCtx, l, id)
 	if err != nil {
-		return "Failed to run container", 1, errors.Wrap(err, "DOCKER RUN")
+		return pb.ActionState_ACTION_IN_PROGRESS, errors.Wrap(err, "DOCKER RUN")
 	}
 
 	failedActionStatus := make(chan pb.ActionState)
 
 	//capturing logs of action container in a go-routine
 	go captureLogs(ctx, id)
 
-	status, err, werr := waitContainer(timeCtx, id)
-	if werr != nil {
-		rerr := removeContainer(ctx, id)
+	status, err := waitContainer(timeCtx, id)
+	if err != nil {
+		rerr := removeContainer(ctx, l, id)
 		if rerr != nil {
-			log.WithField("container_id", id).Errorln("Failed to remove container as ", rerr)
+			l.With("containerId", id)
+			l.Error(errors.Wrapf(rerr, errRemoveContainer))
 		}
-		return "Failed to wait for completion of action", status, errors.Wrap(err, "DOCKER_WAIT")
+		return status, errors.Wrap(err, "DOCKER_WAIT")
 	}
-	rerr := removeContainer(ctx, id)
+	rerr := removeContainer(ctx, l, id)
 	if rerr != nil {
-		return "Failed to remove container of action", status, errors.Wrap(rerr, "DOCKER_REMOVE")
+		return status, errors.Wrap(rerr, "DOCKER_REMOVE")
 	}
-	log.Infoln("Container removed with Status ", pb.ActionState(status))
+	l = l.With(infoContainerRemoved, status.String())
```

Comment:
infoContainerRemoved ("container removed with status") does not seem like a good key instead it seems like the message you would provide to `Info()`. This log entry should instead be done like:
```go
l.With("status", status.String()).Info("container removed")
```